helix-term/src/application.rs

```diff
@@ -1003,32 +1022,21 @@ impl Application {
                         }))
                     }
                     Ok(MethodCall::WorkspaceFolders) => {
-                        let language_server =
-                            self.editor.language_servers.get_by_id(server_id).unwrap();
-
-                        Ok(json!(language_server.workspace_folders()))
+                        Ok(json!(language_server!().workspace_folders()))
                     }
                     Ok(MethodCall::WorkspaceConfiguration(params)) => {
+                        let language_server = language_server!();
                         let result: Vec<_> = params
                             .items
                             .iter()
                             .map(|item| {
-                                let mut config = match &item.scope_uri {
-                                    Some(scope) => {
-                                        let path = scope.to_file_path().ok()?;
-                                        let doc = self.editor.document_by_path(path)?;
-                                        doc.language_config()?.config.as_ref()?
-                                    }
-                                    None => self
-                                        .editor
-                                        .language_servers
-                                        .get_by_id(server_id)
-                                        .unwrap()
-                                        .config()?,
-                                };
+                                let mut config = language_server.config()?;
```

Comment:
Why did you remove the `scope_uri` handeling here?  Was this just an accident while rebasing or on purpose?

Reply:
See here: https://github.com/helix-editor/helix/pull/2507#discussion_r980533113

Reply:
Thanks, I missed that. I just glanced at this after your comment and it looked odd that it was removed but this makes total sense.